File: smash-ranks-android/app/src/main/java/com/garpr/android/models/LiteTournament.java
package com.garpr.android.models;

import android.os.Parcel;
import android.os.Parcelable;
import android.support.annotation.NonNull;

public class LiteTournament extends AbsTournament implements Parcelable {

    public LiteTournament() {
        // only here for GSON purposes
    }

    public LiteTournament(@NonNull final String id, @NonNull final String name,
            @NonNull final SimpleDate date) {
        mId = id;
        mName = name;
        mDate = date;
    }

    @Override
    public Kind getKind() {
        return Kind.LITE;
    }

    public static final Creator<LiteTournament> CREATOR = new Creator<LiteTournament>() {
        @Override
        public LiteTournament createFromParcel(final Parcel source) {
            final LiteTournament lt = new LiteTournament();
            lt.readFromParcel(source);
            return lt;
        }

        @Override
        public LiteTournament[] newArray(final int size) {
            return new LiteTournament[size];
        }
    };

}

File: smash-ranks-android/app/src/main/java/com/garpr/android/models/LiteTournament.kt
package com.garpr.android.models

import android.os.Parcelable
import com.garpr.android.extensions.createParcel

class LiteTournament(
        regions: List<String>? = null,
        date: SimpleDate,
        id: String,
        name: String
) : AbsTournament(
        regions,
        date,
        id,
        name
), Parcelable {

    companion object {
        @JvmField
        val CREATOR = createParcel { LiteTournament(it.createStringArrayList(),
                it.readParcelable(SimpleDate::class.java.classLoader), it.readString(),
                it.readString()) }
    }

    override val kind
        get() = Kind.LITE

}
